packages/quick_actions/quick_actions_ios/ios/Classes/ShortcutStateManaging.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import UIKit
+
+/// Manages the shortcut related states.
+protocol ShortcutStateManaging {
+
+  /// Sets the list of shortcut items.
+  ///
+  /// - Parameter items the list of shortcut items to be parsed and set.
+  func setShortcutItems(_ items: [[String: Any]])
+}
+
+/// A default implementation of `ShortcutStateManaging` protocol.
+final class DefaultShortcutStateManager: ShortcutStateManaging {
+
+  private let service: ShortcutItemServicing
+
+  /// Creates a ShortcutStateManager
+  /// - Parameter service a shortcut item service, with `UIApplication.shared` as default.
+  init(service: ShortcutItemServicing = UIApplication.shared) {
+    self.service = service
+  }
+
+  func setShortcutItems(_ items: [[String: Any]]) {
+    service.shortcutItems = items.compactMap { deserializeShortcutItem(with: $0) }
+  }
+
+  private func deserializeShortcutItem(with serialized: [String: Any]) -> UIApplicationShortcutItem
+  {
+
+    let icon = (serialized["icon"] as? String).map {
+      UIApplicationShortcutIcon(templateImageName: $0)
+    }
+
+    // type and localizedTitle are required.
+    return UIApplicationShortcutItem(
+      type: serialized["type"] as! String,
```

Comment:
Maybe we should not let this plugin crash the production app? Maybe use `Assert` to crash debug app and return nil if either serialized["type"] or serialized["localizedTitle"] is nil in release.

Does the original Objective-C API crashes the App if type or localizedTitle is nil? If so, I'm ok with keep the same behavior and follow up with a fix.

Reply:
> Does the original Objective-C API crashes the App if type or localizedTitle is nil?

I think so, the API is non-null (and that's why it bridges to non-null String in Swift): https://developer.apple.com/documentation/uikit/uiapplicationshortcuticon/1623367-iconwithtemplateimagename?language=objc

Reply:
Adopting pigeon would solve this.

Reply:
Does the Objective-C API _actually_ crash if you pass in `nil` though?  A lot of Objective-C APIs that predate nullability do handle `nil`s under the covers, especially since most implementation files aren't wrapped in `NS_ASSUME_NULL` to catch the error in analysis.  Ideally passing in the wrong types would give an actionable error, but at the least this should introduce a _new_ crash?

Reply:
Good point! Let me try it out. 

The [dart counterpart](https://github.com/flutter/plugins/blob/main/packages/quick_actions/quick_actions_platform_interface/lib/types/shortcut_item.dart) also guarantees that this is never null though. So it likely doesn't matter anyways.

Reply:
I just tried it out - indeed the objc API did not crash - instead it just ignores it. 

Although the [dart counterpart](https://github.com/flutter/plugins/blob/main/packages/quick_actions/quick_actions_platform_interface/lib/types/shortcut_item.dart) guarantees that this is never null, it's probably still good to preserve the exact logic when migrating.